rust/osm2lanes/src/transform/tags_to_lanes/lane_count.rs

```diff
@@ -0,0 +1,375 @@
+
+        // Calculate the both_ways lanes.
+        let both_ways = match (tagged_both_ways, centre_turn_lane.0) {
+            (Some(bw), None) => Infer::Direct(bw),
+            (Some(bw), Some(ctl)) => {
+                // TODO what if the values conflict but are not Direct? Might not ever happen.
+                if (bw > 0 && !ctl) || (bw == 0 && ctl) {
+                    warnings.push(TagsToLanesMsg::ambiguous_tags(
+                        tags.subset(&[LANES + "both_ways", CENTRE_TURN_LANE]),
+                    ));
+                }
+                Infer::Direct(bw)
+            },
+            (None, Some(true)) => Infer::Calculated(1),
+            (None, Some(false)) => Infer::Calculated(0),
+            (None, None) => Infer::Default(0),
+        };
+        let both_way_lanes = both_ways.some().unwrap_or(0);
+
+        let counts = if road.oneway.into() {
+            // Ignore lanes:{both_ways,backward}=
+            if tagged_backward.is_some()
+                || tagged_both_ways.is_some()
+                || centre_turn_lane.0.is_some()
+            {
+                log::trace!("both_ways={both_ways:?} tagged_backward={tagged_backward:?}");
+                warnings.push(TagsToLanesMsg::ambiguous_tags(tags.subset(&[
+                    ONEWAY,
+                    LANES + "both_ways",
+                    LANES + "backward",
+                    CENTRE_TURN_LANE,
+                ])));
+            }
+
+            if let (Some(l), Some(f)) = (tagged_lanes, tagged_forward) {
+                // TODO What is the right warning for straight up conflicts in tag values?
+                if l != f {
+                    warnings.push(TagsToLanesMsg::ambiguous(
+                        &format!("{l}!={f}"),
+                        tags.subset(&[ONEWAY, LANES, LANES + "forward"]),
+                    ));
+                }
+            }
+
+            let assumed_forward = 1; // TODO depends on highway tag
+            Self {
+                lanes: tagged_lanes.map_or_else(
+                    || tagged_forward.map_or(Infer::Default(assumed_forward), Infer::Direct),
+                    Infer::Direct,
+                ),
+                forward: tagged_forward.map_or_else(
+                    || {
+                        tagged_lanes.map_or(Infer::Default(assumed_forward), |l| {
+                            Infer::Direct(l.checked_sub(pre_backward).unwrap())
+                        })
+                    },
+                    Infer::Direct,
+                ),
+                backward: tagged_backward.map_or(Infer::Default(pre_backward), Infer::Direct),
+                both_ways,
+            }
+        } else {
+            // Not oneway
+            match (tagged_lanes, tagged_forward, tagged_backward) {
+                (Some(l), Some(f), Some(b)) => {
+                    if l != f
+                        .checked_add(b)
```

Comment:
Why `checked_add` and an unwrap? Are these just small `usize`s?

Reply:
yeah, clippy doesn't like raw `+`s, but I think those are better than this, so I will revert